### PR TITLE
Xml attributes

### PR DESCRIPTION
**Pull request by kirkeaton**
1. Added output / parsing of XML attributes.

### CODE REVIEW

File: src/foam/core/XML.js [hunk 1]
```diff
@@ -181,6 +186,16 @@ foam.CLASS({
       return this;
     },
 
+    function outputAttributes(v) {
+      var attributes = v.cls_.getAxiomsByClass(foam.core.Property).filter(function (p) { return p.xmlAttribute });
+      if ( attributes.length === 0 ) return this;
+
+      for ( var i = 0; i < attributes.length; i++ ) {
```
Comment (kgrgreer) [inline src/foam/core/XML.js]: spaces before ;'s

File: src/foam/core/XML.js [hunk 2]
```diff
@@ -224,9 +239,24 @@ foam.CLASS({
           }
         },
         FObject: function(v, p) {
-          this.start('<' + this.propertyName(p) + '>');
-          this.output(p.toXML(v, this));
-          this.end('</' +  this.propertyName(p) + '>');
+          if ( v.xmlValue ) {
+            // if v.xmlValue exists then we have attributes
```
Comment (kgrgreer) [inline src/foam/core/XML.js:243]: Is this true? Aren't the use of attributes and a default xmlValue independent? Couldn't you have attributes with tags or a default xmlValue without attributes?

File: src/foam/core/XML.js (hunk 1 above)
Comment (kirkeaton) [reply src/foam/core/XML.js]: Done

File: src/foam/core/XML.js (hunk 2 above)
Comment (kirkeaton) [reply src/foam/core/XML.js:243]: I suppose you could have an xmlValue property without any attributes but can you think of the use case for that? You would be needlessly wrapping the true value in an extra class.

Comment (kgrgreer) [reply src/foam/core/XML.js:243]: <font size="+1">blah blah blah</font>, but I guess we're not going to be modelling HTML (actually, I suppose we do already model XML with U2), but we can leave it until we ever need it.

**Pull request merged**